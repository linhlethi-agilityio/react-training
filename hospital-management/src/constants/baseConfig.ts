import { ROUTES } from './router';

export const NAVIGATION_ITEMS = [
  {
    id: 1,
    name: 'Appointments',
    path: ROUTES.APPOINTMENTS,
  },
  {
    id: 2,
    name: 'Consult Doctors',
    path: ROUTES.CONSULT_DOCTORS,
  },
  {
    id: 3,
    name: 'Diagnostics',
    path: ROUTES.DIAGNOSTICS,
  },
];
