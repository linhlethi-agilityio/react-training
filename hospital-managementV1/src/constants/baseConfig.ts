export const NAVIGATION_ITEM = [
  {
    id: 1,
    name: 'Appointments',
    path: '/appointments',
  },
  {
    id: 2,
    name: 'Consult Doctors',
    path: '/consult-doctors',
  },
  {
    id: 3,
    name: 'Diagnostics',
    path: '/diagnostics',
  },
];

export const QUICK_LINKS = [
  {
    id: 1,
    title: 'Quick Links',
    links: [
      { id: 1, name: 'How it works', path: '#' },
      { id: 2, name: 'About us', path: '#' },
      { id: 3, name: 'Membership', path: '#' },
    ],
  },
  {
    id: 2,
    title: 'Departments',
    links: [
      { id: 1, name: 'Eye care', path: '#' },
      { id: 2, name: 'Cardiac are', path: '#' },
      { id: 3, name: 'Heart care', path: '#' },
    ],
  },
  {
    id: 3,
    title: 'Eye care',
    links: [
      { id: 2, name: 'General Medecine', path: '#' },
      { id: 3, name: 'Orthopdecics', path: '#' },
      { id: 4, name: 'Heart care', path: '#' },
    ],
  },
  {
    id: 4,
    title: 'Customer Care',
    links: [
      { id: 1, name: 'About Us', path: '#' },
      { id: 2, name: 'Contact Us', path: '#' },
      { id: 3, name: 'Send Details', path: '#' },
    ],
  },
];

export const SECTIONS_INFO = {
  SPECIALTY: {
    title: 'Select your Specialty',
    description: 'Private online consultations with verified doctors in all specialists',
  },
  DOWNLOAD_APP: {
    title: 'Download our App',
    description:
      'Access video consultation with Indiaâ€™s top doctors on the Swasthu mobile Application. Connect with doctors online from the comfort of your home.',
  },
  APP_DESCRIPTION: {
    title: 'Why Swasthu?',
    description: 'Lorem ipsum de color iss slsysos kshss sshdusk hbnjfyo ckdhdoi',
  },
  FEEDBACK: {
    title: 'What our user say about us',
    description: 'Lorem ipsum de color iss slsysos kshss sshdusk hbnjfyo ckdhdoi',
  },
};

export const FEEDBACKS = [
  'Very easy to book,maintain history. Hassle free from older versions of booking appointment via telephone.. Thanks Practo for making it simple.',
  'Very easy to book,maintain history. ersions of  via telephone.. Thanks Practo for making it simple.',
  'Very easy to book,maintain history. Hversions of booking appointment or making assle free from older  it simple.',
];

export const APP_DESCRIPTIONS = [
  {
    id: '1',
    title: 'Curated Docotrs',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Urna gravida id lobortis eget tristique ultrices feugiat neque elit. Amet imperdiet donec cras habitant ultrices sed in mi scelerisque. Maecenas tempus amet aenean in eu ullamcorper viverra pulvinar. Nibh et elementum parturient viverra in. In morbi vehicula dignissim pharetra, tristique. ',
  },
  {
    id: '2',
    title: 'Curated Docotrs Docotrs',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Urna gravida id lobortis eget tristique ultrices feugiat neque elit. Amet imperdiet donec cras habitant ultrices sed in mi scelerisque. Maecenas tempus amet aenean in eu ullamcorper viverra pulvinar. Nibh et elementum parturient viverra in. In morbi vehicula dignissim pharetra, tristique. ',
  },
  {
    id: '3',
    title: 'Curated Docotrs Docotrs',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Urna gravida id lobortis eget tristique ultrices feugiat neque elit. Amet imperdiet donec cras habitant ultrices sed in mi scelerisque. Maecenas tempus amet aenean in eu ullamcorper viverra pulvinar. Nibh et elementum parturient viverra in. In morbi vehicula dignissim pharetra, tristique. ',
  },
];

export const DOWNLOAD_APP = {
  iosAppLink: 'https://www.apple.com/vn/app-store/',
  androidAppLink: 'https://play.google.com/store/games',
};

export const USER_NAVBAR_ITEMS = [
  'My Profile',
  'Appointment History',
  'Diagnostics Result',
  'Upcoming Appointments',
  'Others',
];
