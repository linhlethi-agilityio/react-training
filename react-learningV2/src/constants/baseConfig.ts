export const PERSON = {
  name: 'Gregorio Y. Zara',
  theme: {
    backgroundColor: 'black',
    color: 'pink',
  },
};

export const TODOS = [
  {
    id: '1',
    name: 'todo 1',
  },
  {
    id: '2',
    name: 'todo 2',
  },
  {
    id: '3',
    name: 'todo 3',
  },
];

export const SCULPTURE_LIST = [
  {
    name: 'Homenaje a la Neurocirug√≠a',
    artist: 'Marta Colvin Andrade',
    description:
      'Although Colvin is predominantly known for abstract themes that allude to pre-Hispanic symbols, this gigantic sculpture, an homage to neurosurgery, is one of her most recognizable public art pieces.',
    url: 'https://i.imgur.com/Mx7dA2Y.jpg',
    alt: 'A bronze statue of two crossed hands delicately holding a human brain in their fingertips.',
  },
  {
    name: 'Moai',
    artist: 'Unknown Artist',
    description:
      'Located on the Easter Island, there are 1,000 moai, or extant monumental statues, created by the early Rapa Nui people, which some believe represented deified ancestors.',
    url: 'https://i.imgur.com/RCwLEoQm.jpg',
    alt: 'Three monumental stone busts with the heads that are disproportionately large with somber faces.',
  },
  {
    name: 'Blue Nana',
    artist: 'Niki de Saint Phalle',
    description:
      'The Nanas are triumphant creatures, symbols of femininity and maternity. Initially, Saint Phalle used fabric and found objects for the Nanas, and later on introduced polyester to achieve a more vibrant effect.',
    url: 'https://i.imgur.com/Sd1AgUOm.jpg',
    alt: 'A large mosaic sculpture of a whimsical dancing female figure in a colorful costume emanating joy.',
  },
  {
    name: 'Ultimate Form',
    artist: 'Barbara Hepworth',
    description:
      'This abstract bronze sculpture is a part of The Family of Man series located at Yorkshire Sculpture Park. Hepworth chose not to create literal representations of the world but developed abstract forms inspired by people and landscapes.',
    url: 'https://i.imgur.com/2heNQDcm.jpg',
    alt: 'A tall sculpture made of three elements stacked on each other reminding of a human figure.',
  },
  {
    name: 'Cavaliere',
    artist: 'Lamidi Olonade Fakeye',
    description:
      "Descended from four generations of woodcarvers, Fakeye's work blended traditional and contemporary Yoruba themes.",
    url: 'https://i.imgur.com/wIdGuZwm.png',
    alt: 'An intricate wood sculpture of a warrior with a focused face on a horse adorned with patterns.',
  },
  {
    name: 'Hippos',
    artist: 'Taipei Zoo',
    description: 'The Taipei Zoo commissioned a Hippo Square featuring submerged hippos at play.',
    url: 'https://i.imgur.com/6o5Vuyu.jpg',
    alt: 'A group of bronze hippo sculptures emerging from the sett sidewalk as if they were swimming.',
  },
];
