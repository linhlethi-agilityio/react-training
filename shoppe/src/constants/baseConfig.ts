export const NAVIGATION_ITEM = [
  {
    id: 1,
    name: 'Contact',
    path: '/contact',
  },
  {
    id: 2,
    name: 'Terms or services',
    path: '/terms-or-services',
  },
  {
    id: 3,
    name: 'shipping and returns',
    path: '/shipping-and-returns',
  },
];

export const SOCIAL_INFORMATION = {
  facebook: 'https://www.facebook.com/',
  twitter: 'https://twitter.com/',
  instagram: 'https://www.instagram.com/',
  linkedin: 'https://www.linkedin.com/',
};

export const SORT_OPTIONS = [
  {
    label: 'By Price Ascending',
    value: 'price_asc',
  },
  {
    label: 'By Price Descending',
    value: 'price_desc',
  },
  {
    label: 'By Product Name Ascending',
    value: 'name_asc',
  },
  {
    label: 'By Product Name Descending',
    value: 'name_desc',
  },
];

export const MIN_QUANTITY = 1;
export const MAX_QUANTITY = 10;

export const LIMIT_PRODUCTS = 6;
