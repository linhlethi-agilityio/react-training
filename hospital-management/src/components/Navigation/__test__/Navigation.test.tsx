import { describe, expect, it } from 'vitest';
import { MemoizedNavigation } from '..';
import { render } from '@testing-library/react';
import { NAVIGATION_ITEMS } from '@constants/baseConfig';

describe('testing Navigation component', () => {
  it('snapshot', () => {
    const navigation = render(<MemoizedNavigation items={NAVIGATION_ITEMS} />);

    expect(navigation).toMatchSnapshot();
  });

  it('should render navigation item correct', () => {
    const { getByText } = render(<MemoizedNavigation items={NAVIGATION_ITEMS} />);

    NAVIGATION_ITEMS.map((item) => {
      const text = getByText(item.name);

      expect(text).toBeTruthy();
    });
  });
});
