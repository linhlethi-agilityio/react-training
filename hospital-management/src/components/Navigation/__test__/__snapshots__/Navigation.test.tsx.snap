// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing Navigation component > snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="mantine-14n9mxy"
      >
        <a
          class="mantine-UnstyledButton-root mantine-NavLink-root mantine-1w8njn9"
          href="/appointments"
        >
          <span
            class="mantine-dstkpx mantine-NavLink-body"
          >
            <span
              class="mantine-Text-root mantine-NavLink-label mantine-162lpqd"
            >
              Appointments
            </span>
            <span
              class="mantine-Text-root mantine-NavLink-description mantine-1hwiwhc"
            />
          </span>
        </a>
        <div
          aria-hidden="true"
          class="mantine-1avyp1d"
          style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
        >
          <div
            style="opacity: 0; transition: opacity 200ms ease;"
          >
            <div
              class="mantine-1c04k7q mantine-NavLink-children"
            />
          </div>
        </div>
        <a
          class="mantine-UnstyledButton-root mantine-NavLink-root mantine-1w8njn9"
          href="/consult-doctors"
        >
          <span
            class="mantine-dstkpx mantine-NavLink-body"
          >
            <span
              class="mantine-Text-root mantine-NavLink-label mantine-162lpqd"
            >
              Consult Doctors
            </span>
            <span
              class="mantine-Text-root mantine-NavLink-description mantine-1hwiwhc"
            />
          </span>
        </a>
        <div
          aria-hidden="true"
          class="mantine-1avyp1d"
          style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
        >
          <div
            style="opacity: 0; transition: opacity 200ms ease;"
          >
            <div
              class="mantine-1c04k7q mantine-NavLink-children"
            />
          </div>
        </div>
        <a
          class="mantine-UnstyledButton-root mantine-NavLink-root mantine-1w8njn9"
          href="/diagnostics"
        >
          <span
            class="mantine-dstkpx mantine-NavLink-body"
          >
            <span
              class="mantine-Text-root mantine-NavLink-label mantine-162lpqd"
            >
              Diagnostics
            </span>
            <span
              class="mantine-Text-root mantine-NavLink-description mantine-1hwiwhc"
            />
          </span>
        </a>
        <div
          aria-hidden="true"
          class="mantine-1avyp1d"
          style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
        >
          <div
            style="opacity: 0; transition: opacity 200ms ease;"
          >
            <div
              class="mantine-1c04k7q mantine-NavLink-children"
            />
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="mantine-14n9mxy"
    >
      <a
        class="mantine-UnstyledButton-root mantine-NavLink-root mantine-1w8njn9"
        href="/appointments"
      >
        <span
          class="mantine-dstkpx mantine-NavLink-body"
        >
          <span
            class="mantine-Text-root mantine-NavLink-label mantine-162lpqd"
          >
            Appointments
          </span>
          <span
            class="mantine-Text-root mantine-NavLink-description mantine-1hwiwhc"
          />
        </span>
      </a>
      <div
        aria-hidden="true"
        class="mantine-1avyp1d"
        style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
      >
        <div
          style="opacity: 0; transition: opacity 200ms ease;"
        >
          <div
            class="mantine-1c04k7q mantine-NavLink-children"
          />
        </div>
      </div>
      <a
        class="mantine-UnstyledButton-root mantine-NavLink-root mantine-1w8njn9"
        href="/consult-doctors"
      >
        <span
          class="mantine-dstkpx mantine-NavLink-body"
        >
          <span
            class="mantine-Text-root mantine-NavLink-label mantine-162lpqd"
          >
            Consult Doctors
          </span>
          <span
            class="mantine-Text-root mantine-NavLink-description mantine-1hwiwhc"
          />
        </span>
      </a>
      <div
        aria-hidden="true"
        class="mantine-1avyp1d"
        style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
      >
        <div
          style="opacity: 0; transition: opacity 200ms ease;"
        >
          <div
            class="mantine-1c04k7q mantine-NavLink-children"
          />
        </div>
      </div>
      <a
        class="mantine-UnstyledButton-root mantine-NavLink-root mantine-1w8njn9"
        href="/diagnostics"
      >
        <span
          class="mantine-dstkpx mantine-NavLink-body"
        >
          <span
            class="mantine-Text-root mantine-NavLink-label mantine-162lpqd"
          >
            Diagnostics
          </span>
          <span
            class="mantine-Text-root mantine-NavLink-description mantine-1hwiwhc"
          />
        </span>
      </a>
      <div
        aria-hidden="true"
        class="mantine-1avyp1d"
        style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
      >
        <div
          style="opacity: 0; transition: opacity 200ms ease;"
        >
          <div
            class="mantine-1c04k7q mantine-NavLink-children"
          />
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
